### DISCUSSION TITLE
tutorials/intro-strawberry/08-rest-datasource

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#131] [course feedback]
# tutorials/intro-strawberry/08-rest-datasource

Set up a REST API as a data source using an auto-generated HTTPClient.

https://www.apollographql.com/tutorials/intro-strawberry/08-rest-datasource

**Comment by akristen** [comment]
OK! Wanted to clarify something if other people get to this point and get stuck. I'm not a dev or engineer, but actually technical writer learning about GraphQL. If you're experienced, this might seem obvious but this is my first time working in Python, hehe. 

If you're working on Windows, `source` won't work. I recommend reviewing Python docs to find the command that fits your use case: https://docs.python.org/3/library/venv.html

There's a chance that the localhost might fail to connect in this step. I'm not sure why and maybe it was just for me. If you're seeing (venv) but localhost isn't loading, go back to "Overview and setup" and repeat those steps. It should work after that :)

**Reply by mabuyo** [reply]
Thanks for the tip @akristen - I hope this is helpful for other Windows users!

Curious if you saw that we had other languages available for Intro to GraphQL courses? We have JavaScript, TypeScript, Java and C#. If you're more familiar with any of those languages, you might enjoy those more!

You can also check out https://graphql.com/learn/ for language-agnostic material :)

**Comment by RomainBsb** [comment]
Hi, I got this error when using openapi-python-client. Any idea how to solve it? 
FileExistsError: [Errno 17] File exists: '/Users/.../odyssey-intro-strawberry/mock_spotify_rest_api_client'
(.venv) ➜  odyssey-intro-strawberry git:(main) ✗ openapi-python-client generate --meta none --path data/openapi.json

Generating mock_spotify_rest_api_client
Error(s) encountered while generating, client was not created

ruff failed

models/spotify_object_authorization_code_credentials.py:62:22: F821 Undefined name `cast`
models/spotify_request_query_params_get_mefollowing.py:48:16: F821 Undefined name `cast`
models/spotify_request_query_params_get_search.py:69:28: F821 Undefined name `cast`
Found 1936 errors (1933 fixed, 3 remaining).

**Reply by RomainBsb** [reply]
Seems that the client was still created despite the message. Indeed I was able to run the next steps of this tutorial

**Reply by mabuyo** [reply]
Glad to hear that @RomainBsb - that's a misleading error. We'll keep an eye out for anyone that runs into the same issue!

**Reply by JasonLi314** [reply]
@RomainBsb I think newest version of openapi (`0.21.5` at this time) has this issue. I was able to fix the issue by downgrading back to `0.19.0`, the version this tutorial used, using this command: `pipx inject openapi-python-client "openapi-python-client==0.19.0" --include-deps --force`

my env specs: 
* python version: 3.12.0
* pipx version: 1.7.1
* macos

**Reply by AllanAtNephos** [reply]
I had this one as well.  I hadn't noticed it still works despite the error, so I'm glad I checked here!  I was just about to add cast to the list of imported functions from the typing library in the two files that were raising the error (which would have dealt with the specific error?) - I will just ignore it!

**Comment by mjosipovic** [comment]
Hello, I missed from where to get openapi.json. Please help

**Reply by mjosipovic** [reply]
I am sorry  for confusion. I found the link to the github repo in the Overview and setup lesson.  Thanks a lot for providing this great tutorial.
